lib/components/viewers/route-viewer.js

```diff
@@ -66,52 +67,27 @@ class RouteViewer extends Component {
   onSearchChange = (event) => {
     const { target } = event
     const { value } = target
-    this.setState({ search: value })
+    this.props.setRouteViewerFilter({ search: value })
   };
 
   render () {
     const {
+      agencies,
+      filter,
       findRoute,
       hideBackButton,
-      languageConfig,
+      intl,
+      modes,
       routes,
       setViewedRoute,
       transitOperators,
       viewedRoute
     } = this.props
-    const { agency, mode, search } = this.state
-    const routesArray = routes ? Object.values(routes) : []
+    const { search } = filter
 
-    const filteredAndSortedRoutes = routesArray
-      .filter(
-        (route) =>
-          // If the filter isn't defined, don't check.
-          (!agency || agency === route.agencyName) &&
-          (!mode || mode === route.mode) &&
-          // If user search is active, filter here
-          (!search ||
-            ((route.longName && route.longName.includes(search)) ||
-              (route.shortName && route.shortName.includes(search))))
-      )
-      .sort(coreUtils.route.makeRouteComparator(transitOperators))
-
-    // get agency and mode lists from route list
-    const agencies = Array.from(
-      new Set(routesArray.map((route) => route.agencyName || route.agency.name))
-    )
-      .filter((agency) => agency !== undefined)
-      .sort()
-
-    const modes = Array.from(
-      new Set(
-        routesArray
-          .filter((route) => route.mode !== undefined)
-          .map((route) => route.mode.toLowerCase())
-      )
+    const sortedRoutes = routes.sort(
+      coreUtils.route.makeRouteComparator(transitOperators)
     )
```

Comment:
Can this sort functionality live in the selector, too?

Reply:
It very much can! Moved in ec7e88565eba513e9799ecc01296f122ac9a937f